src/supernova/Readme.md

```diff
@@ -306,8 +306,35 @@ After any number of iterations of `prove_step`, we can check that the current pr
 To verify that $(z_i, z'_i)$ are correct, the verifier needs to recompute the public outputs of the latest proof $u'$. Since this is the output on the secondary curve, the first entry $u'.X_0$ will be the output of the primary curve circuit producing $(\mathsf{pc}_i, z_i)$ and the accumulator $U'$ in which we will fold $u'$. The second entry $u'.X_1$ is the output of the last circuit on the secondary curve, which will have folded the proof for $(\mathsf{pc}_i, z_i)$ into $U[\ ]$.
 
 - $u'.X_0 \stackrel{?}{=} H(\mathsf{vk}, i, \mathsf{pc}_i, z_0, z_i, U')$
-- $u'.X_1 \stackrel{?}{=} H'(\mathsf{vk}, i, z'_0, z'_i, U[\ ])$ 
+- $u'.X_1 \stackrel{?}{=} H'(\mathsf{vk}, i, z'_0, z'_i, U[\ ])$
 
 We then verify that $(u',w')$ is a satisfying circuit, which proves that all relaxed instances $U[\ ], U'$ were correctly updated through by folding proof.
 
-We then need to verify that all accumulators $(U[\ ], W[\ ])$ and $(U', W')$ are correct by checking the circuit satisfiability. 
\ No newline at end of file
+We then need to verify that all accumulators $(U[\ ], W[\ ])$ and $(U', W')$ are correct by checking the circuit satisfiability.
+
+## Links to relevant
+
+### [RecursiveSNARK](./mod.rs#L400)
+
+1) [`new`](./mod.rs#L443)
+2) [`prove_step`](./mod.rs#L615)
+3) [`verify`](./mod.rs#L797)
+
+### [SuperNova Augmented Circuit](./circuit.rs#L126)
+
+1) [`synthesize_base_case`](./circuit.rs#L276)
+2) [`synthesize_non_base_case`](./circuit.rs#L323)
+3) [`synthesize`](./circuit.rs#L410)
+4) [Public IO](./circuit.rs#L584)
+
+### [SuperNova CompressedSNARK](./snark.rs)
+
+1) [`prove`](./snark.rs#L142)
+2) [`verify`](./snark.rs#L227)
+
+#### [BatchedRelaxedR1CSSNARKTrait](../traits/snark.rs#L65)
+
+1) [Batched Spartan](../spartan/batched.rs)
+2) [Batched Spartan with computation commitments](../spartan/batched_ppsnark.rs)
+3) [Batched sumcheck](../spartan/sumcheck.rs#L464)
+4) [Batched sumcheck verify](../spartan/sumcheck.rs#L65)
```

Comment:
Those links will bit-rot extremely: every `cargo fmt` will make the whole file's set of links become obsolete. 

I think the only way to have something maintainable is to have an outline as part of a description of a protocol (e.g. here), and have *exactly the string used to define those titles and subtitles* reproduced in a code comment at the appropriate place.

E.g. `#### Branch: i>0 `synthesize_non_base_case`, which could feature in a code comment.